Snowflake.Data.Tests/Mock/MockServiceName.cs

```diff
@@ -0,0 +1,107 @@
+﻿/*
+ * Copyright (c) 2012-2019 Snowflake Computing Inc. All rights reserved.
+ */
+
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Net.Http;
+
+namespace Snowflake.Data.Tests.Mock
+{
+    using Snowflake.Data.Core;
+
+    class MockServiceName : IRestRequester
+    {
+        public const string INIT_SERVICE_NAME = "init";
+        public Task<T> PostAsync<T>(IRestRequest request, CancellationToken cancellationToken)
+        {
+            return PrepareResponse<T>(HttpMethod.Post, request, cancellationToken);
+        }
+
+        public T Post<T>(IRestRequest postRequest)
+        {
+            return Task.Run(async () => await PostAsync<T>(postRequest, CancellationToken.None)).Result;
+        }
+
+        public T Get<T>(IRestRequest request)
+        {
+            return Task.Run(async () => await GetAsync<T>(request, CancellationToken.None)).Result;
+        }
+
+        public Task<T> GetAsync<T>(IRestRequest request, CancellationToken cancellationToken)
+        {
+            return PrepareResponse<T>(HttpMethod.Get, request, cancellationToken);
```

Comment:
I don't think in your Mock you need this. Get to snowflake server only happened in long running query, where client is asking for query progress/result. In your mock, you always assume POST. Also, in line 44 in this class, you hard code HttpMethod to Post, so I would just return a null response here.

Reply:
Addressed